paddle/phi/core/string_tensor.cc

```diff
@@ -141,13 +142,15 @@ void* StringTensor::AllocateFrom(Allocator* allocator,
           "The meta data must be valid when call the mutable data function."));
   size_t bytes = numel() * SizeOf(this->dtype());
   if (requested_size) {
-    PADDLE_ENFORCE_GE(requested_size,
-                      bytes,
-                      errors::InvalidArgument(
-                          "The reserved size %d should be enough to meet the "
-                          "volume required by metadata %d.",
-                          requested_size,
-                          bytes));
+    if (check_size) {
+      PADDLE_ENFORCE_GE(requested_size,
+                        bytes,
+                        errors::InvalidArgument(
+                            "The reserved size %d should be enough to meet the "
+                            "volume required by metadata %d.",
+                            requested_size,
+                            bytes));
+    }
     bytes = requested_size;
```

Comment:
In dense_tensor.cc, when `requested_size=0` and `!check_size`, the `bytes` is compulsorily assigned as `0`. However, the similar codes are not seen in string_tensor.cc.

Reply:
Yes, only DenseTensor is supported currently. So, other tensors are not changed.